sdks/java/io/elasticsearch-tests/elasticsearch-tests-common/src/test/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIOTestCommon.java

```diff
@@ -197,8 +191,18 @@ void testWriteWithErrors() throws Exception {
     List<String> input =
         ElasticSearchIOTestUtils.createDocuments(
             numDocs, ElasticSearchIOTestUtils.InjectionMode.INJECT_SOME_INVALID_DOCS);
-    expectedException.expect(isA(IOException.class));
-    expectedException.expectMessage(
+    // need to avoid flow interruption to close the DoFnTester
+    Exception raisedException = null;
+    try {
+      // inserts into Elasticsearch
+      fnTester.processBundle(input);
+    } catch (IOException exception) {
+      raisedException = exception;
+    }
+    fnTester.finishBundle();
```

Comment:
* finishBundle shouldn't be necessary - processBundle should have called it. Does it not work without this?
* Can you use try-with-resources for closing (try (DoFnTester...))?

Reply:
- indeed it works without the `finishBundle`, thanks for pointing out
- yes better indeed to use try with resources. And it is compatible with `ExpectedException` junit rule, thanks.